cpp/tests/test_toppra_approach.cpp

```diff
@@ -0,0 +1,152 @@
+// Author: Niels Dehio, 13. September 2020
+#include <toppra/algorithm/toppra.hpp>
+#include <toppra/constraint/linear_joint_acceleration.hpp>
+#include <toppra/constraint/linear_joint_velocity.hpp>
+#include <toppra/geometric_path/piecewise_poly_path.hpp>
+#include <toppra/parametrizer/const_accel.hpp>
+#include <toppra/toppra.hpp>
+
+#include "gtest/gtest.h"
+
+#define RTOL 0.01
+#define ATOL 0.0
+
+// This cpp-example implements the python-example described here:
+// https://hungpham2511.github.io/toppra/quickstart.html
+class Approach : public testing::Test {
+ public:
+  Approach(){};
+
+  void formatVecToMat(const std::vector<Eigen::VectorXd,
+                                        Eigen::aligned_allocator<Eigen::VectorXd>>& vec,
+                      Eigen::MatrixXd& mat) {
+    mat.resize(vec.at(0).rows(), vec.size());
+    for (size_t i = 0; i < vec.size(); i++) {
+      mat.col(i) = vec.at(i);
+    }
+  }
+};
+
+TEST_F(Approach, ToppraCompleteExample) {
+  const bool printInfo = false;
+
+  //#### create piecewise polynomial geometric path ####
+  const int numJoints = 2;
+  toppra::Vector position0{numJoints}, position1{numJoints}, position2{numJoints};
+  position0 << 0.0, 0.0;
+  position1 << 1.0, 2.0;
+  position2 << 2.0, 0.0;
+  toppra::Vector velocity0{numJoints}, velocity1{numJoints}, velocity2{numJoints};
+  velocity0 << 0.0, 0.0;
+  velocity1 << 1.0, 1.0;
+  velocity2 << 0.0, 0.0;
+  toppra::Vectors positions = {position0, position1,
+                               position2};  //[(0, 0), (1, 2), (2, 0)]
+  toppra::Vectors velocities = {velocity0, velocity1,
+                                velocity2};  //[(0, 0), (1, 1), (0, 0)]
+
+  std::vector<toppra::value_type> steps;
+  steps = std::vector<toppra::value_type>{0, 1, 2};
+  toppra::PiecewisePolyPath hermite =
+      toppra::PiecewisePolyPath::constructHermite(positions, velocities, steps);
+  toppra::GeometricPathPtr path;
+  path = std::make_shared<toppra::PiecewisePolyPath>(hermite);
+
+  //#### create linear joint-space constraints ####
+  toppra::Vector velLimitLower = Eigen::VectorXd::Zero(numJoints);
+  toppra::Vector velLimitUpper = Eigen::VectorXd::Zero(numJoints);
+  toppra::Vector accLimitLower = Eigen::VectorXd::Zero(numJoints);
+  toppra::Vector accLimitUpper = Eigen::VectorXd::Zero(numJoints);
+  velLimitLower << -1, -0.5;
+  velLimitUpper << 1, 0.5;
+  accLimitLower << -0.05, -0.1;
+  accLimitUpper << 0.2, 0.3;
+
+  toppra::LinearConstraintPtr ljv, lja;
+  ljv = std::make_shared<toppra::constraint::LinearJointVelocity>(
+      velLimitLower, velLimitUpper);  //[[-1, 1], [-0.5, 0.5]]
+  lja = std::make_shared<toppra::constraint::LinearJointAcceleration>(
+      accLimitLower, accLimitUpper);  //[[-0.05, 0.2], [-0.1, 0.3]]
+  lja->discretizationType(toppra::DiscretizationType::Interpolation);
```

Comment:
@ndehio This is necessary.

Reply:
Thanks a lot for completing my pull request. What about adding also `ljv->discretizationType(toppra::DiscretizationType::Interpolation);`  ?